snuba/datasets/entities/sessions.py

```diff
@@ -79,6 +104,131 @@ def plus_columns(
+terminal_status = in_condition(status, [Literal(None, status) for status in [2, 3, 4]])
+
+# These here are basically the same statements as the matview query
+sessions_raw_translators = TranslationMappers(
+    columns=[
+        ColumnToCurriedFunction(
+            None,
+            "duration_quantiles",
+            FunctionCall(None, "quantilesIf", quantiles),
+            (duration, duration_condition),
+        ),
+        ColumnToFunction(None, "duration_avg", "avgIf", (duration, duration_condition)),
+        ColumnToFunction(
+            None, "sessions", "sumIf", (quantity, eq(seq, Literal(None, 0))),
+        ),
+        ColumnToFunction(
+            None, "sessions_crashed", "sumIf", (quantity, eq(status, Literal(None, 2)))
+        ),
+        ColumnToFunction(
+            None, "sessions_abnormal", "sumIf", (quantity, eq(status, Literal(None, 3)))
+        ),
+        ColumnToFunction(
+            None,
+            "sessions_errored",
+            "plus",
+            (
+                FunctionCall(
+                    None,
+                    "uniqIf",
+                    (session_id, _and(has_errors, neq(session_id, lit_nil))),
+                ),
+                FunctionCall(
+                    None,
+                    "sumIf",
+                    (quantity, _and(terminal_status, eq(session_id, lit_nil))),
+                ),
+            ),
+        ),
+        ColumnToFunction(None, "users", "uniqIf", (distinct_id, did_not_nil)),
+        ColumnToFunction(
+            None,
+            "users_crashed",
+            "uniqIf",
+            (distinct_id, _and(eq(status, Literal(None, 2)), did_not_nil)),
+        ),
+        ColumnToFunction(
+            None,
+            "users_abnormal",
+            "uniqIf",
+            (distinct_id, _and(eq(status, Literal(None, 3)), did_not_nil)),
+        ),
+        ColumnToFunction(
+            None,
+            "users_errored",
+            "uniqIf",
+            (distinct_id, _and(has_errors, did_not_nil)),
+        ),
+    ]
+)
+
+
+class SessionsQueryStorageSelector(QueryStorageSelector):
+    def __init__(self) -> None:
+        self.materialized_storage = get_storage(StorageKey.SESSIONS_HOURLY)
+        self.raw_storage = get_storage(StorageKey.SESSIONS_RAW)
+
+    def select_storage(
+        self, query: Query, request_settings: RequestSettings
+    ) -> StorageAndMappers:
+        granularity = extract_granularity_from_query(query, "started") or 3600
+        use_materialized_storage = granularity >= 3600 and (granularity % 3600) == 0
+
+        allow_subhour_sessions = state.get_config("allow_subhour_sessions", 0)
+        if not allow_subhour_sessions:
+            use_materialized_storage = True
+
+        metrics.increment(
+            "query.selector",
+            tags={
+                "granularity": granularity,
+                "selected_storage": "materialized"
+                if use_materialized_storage
+                else "raw",
```

Comment:
This metric collection should go before the killswitch, otherwise you are not logging what the selector would select if the killswitch was on, which is what you want to know before turning the killswitch on.